text/3487-feature-visibility.md

```diff
@@ -0,0 +1,160 @@
+- Feature Name: feature-metadata
+- Start Date: 2023-09-08
+- RFC PR: [rust-lang/rfcs#3487](https://github.com/rust-lang/rfcs/pull/3487)
+- Rust Issue:
+  [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC describes a new key under `features` in `Cargo.toml` to indicate that a
+feature is private.
+
+Please see the parent meta RFC for background information: [`feature-metadata`].
+
+# Motivation
+
+[motivation]: #motivation
+
+WIP
+
+No way to hide unstable API such as testing-only features or internally enabled
+features.
```

Comment:
This needs to be expanded out
- At one point, people had "clippy" features that were just meant for local development
- `winnow` and `clap` have `debug` features that are only meant for local development
- It can help simplify feature graphs to have intermediate hidden features that fan out

Reply:
Done, please feel free to rereview or edit as needed